### PR TITLE
chore: Migrate tests to RTL (5)

### LINKED ISSUE #7045: Migrate tests to RTL (5)
### Component test suites to migrate:
- Page
- Pagination
- Radio
- SearchInput
- Select
- SelectToggle
- SimpleList
- Switch
- TextArea
- TextInput
- TextInputGroup
- TimePicker
- Title
- ToggleGroup
- Wizard
- Bullseye
- Flex

### PR DESCRIPTION
**Pull request by jeffpuzzo**
**What**: Closes #7045

### CODE REVIEW

**Review by thatblindgeye — COMMENTED** [review]
There were some tests that looked like they could be place into their own test file (the SimpleListGroup and SimpleListItem tests at the end of the SimpleList test file, most of the tests in the ToggleGroup test file). I didn't comment on them specifically since those are things that I figured would be part of the next phase.

Below are mainly nitpicks or asking for clarification.

File: packages/react-core/src/components/ToggleGroup/__tests__/ToggleGroup.test.tsx [hunk 1]
```diff
@@ -9,41 +11,43 @@ const props = {
...
-  const view = render(<ToggleGroupItem text="test" selected buttonId="toggleGroupItem" aria-label="basic selected" />);
-  expect(view.container).toMatchSnapshot();
-});
-
-test('basic not selected', () => {
-  const view = render(<ToggleGroupItem text="test" buttonId="toggleGroupItem" aria-label="basic not selected" />);
-  expect(view.container).toMatchSnapshot();
+describe('ToggleGroup', () => {
+  test('basic selected', () => {
+    const { asFragment } = render(<ToggleGroupItem text="test" selected buttonId="toggleGroupItem" aria-label="basic selected" />);
```
[the remark this answers is not in the document]
Comment (jeffpuzzo) [reply packages/react-core/src/components/ToggleGroup/__tests__/ToggleGroup.test.tsx:16]: I agree that is probably what was originally intended. Will update.

File: packages/react-core/src/components/TextArea/__tests__/TextArea.test.tsx [hunk 2]
```diff
@@ -1,112 +1,112 @@
...
+
+  test('read only text input using readOnly', () => {
+    const { asFragment } = render(<TextArea {...props} aria-label="read only textarea" readOnly />);
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  test('invalid text area', () => {
+    const { asFragment } = render(<TextArea {...props} required validated={'error'} aria-label="invalid textarea" />);
+    expect(asFragment()).toMatchSnapshot();
+  });
```
[the remark this answers is not in the document]
Comment (jeffpuzzo) [reply packages/react-core/src/components/TextArea/__tests__/TextArea.test.tsx:51]: Good catch

File: packages/react-core/src/components/Switch/__tests__/Switch.test.tsx [hunk 3]
```diff
@@ -1,97 +1,103 @@
...
+    render(<Switch id="onChange-switch" {...props} aria-label="Switch label" />);
 
-test('should throw console error when no aria-label or label is given', () => {
-  const myMock = jest.fn();
-  global.console = { ...global.console, error: myMock };
-  mount(<Switch {...props} />);
-  expect(myMock).toBeCalled();
-});
+    userEvent.click(screen.getByLabelText('Switch label'));
+    expect(props.onChange.mock.calls[0][0]).toBe(true);
```
[the remark this answers is not in the document]
Comment (jeffpuzzo) [reply packages/react-core/src/components/Switch/__tests__/Switch.test.tsx:69]: No good reason. Just overlooked it. Will update.

File: packages/react-core/src/components/Select/__tests__/SelectToggle.test.tsx [hunk 4]
```diff
@@ -1,148 +1,104 @@
...
-    map.click({ target: document });
-    expect(mockToggle.mock.calls).toHaveLength(0);
-    expect(document.removeEventListener).toHaveBeenCalledWith('click', expect.any(Function));
-  });
 
-  test('on touch outside has been removed', () => {
-    const map = {} as any;
-    document.addEventListener = jest.fn((event, cb) => {
-      map[event] = cb;
+    test('touch on document', () => {
```
[the remark this answers is not in the document]
Comment (jeffpuzzo) [reply packages/react-core/src/components/Select/__tests__/SelectToggle.test.tsx:57]: I don't think it's necessary either.

File: packages/react-core/src/components/Switch/__tests__/Switch.test.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (jeffpuzzo) [reply packages/react-core/src/components/Switch/__tests__/Switch.test.tsx]: Updated these to use `toHaveBeenCalledWith` instead in the event the first argument is meaningful for these tests for whatever reason. Where the 2nd param will vary and contains a lot of details we shouldn't care about.

Example: `expect(props.onChange).toHaveBeenCalledWith(true, expect.any(Object));`

File: packages/react-core/src/components/ToggleGroup/__tests__/ToggleGroup.test.tsx [hunk 5]
```diff
@@ -9,41 +11,43 @@ const props = {
...
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  test('item passes selection and event to onChange handler', () => {
+    render(
+      <ToggleGroupItem text="test" buttonId="toggleGroupItem" onChange={props.onChange} aria-label="onChange handler" />
+    );
+
+    userEvent.click(screen.getByRole('button'));
+    expect(props.onChange).toHaveBeenCalled();
```
Comment (wise-king-sullyman) [inline packages/react-core/src/components/ToggleGroup/__tests__/ToggleGroup.test.tsx:41]: Test is no longer testing arguments being passed here as well.

File: packages/react-core/src/components/Pagination/__tests__/Pagination.test.tsx [hunk 6]
```diff
@@ -1,332 +1,285 @@
...
-        .find('[data-action="last"]')
-        .first()
-        .simulate('click');
-      expect(onSetPage.mock.calls).toHaveLength(0);
+    test('custom pagination toggle', () => {
+      const { asFragment } = render(
+        <Pagination
+          itemCount={40}
+          toggleTemplate={'${firstIndex} - ${lastIndex} - ${itemCount} - ${itemsTitle}'}
+          data-testid="test-id"
```
[the remark this answers is not in the document]
Comment (jeffpuzzo) [reply packages/react-core/src/components/Pagination/__tests__/Pagination.test.tsx:81]: Just leftover from when I wasn't using `asFragment`.

File: packages/react-core/src/components/TextArea/__tests__/TextArea.test.tsx [hunk 7]
```diff
@@ -1,112 +1,107 @@
...
-  const myMock = jest.fn();
-  global.console = { ...global.console, error: myMock };
-  mount(<TextArea {...props} aria-label="test textarea" />);
-  expect(myMock).not.toBeCalled();
+describe('TextArea', () => {
+  test('textarea input passes value and event to onChange handler', () => {
+    render(<TextAreaBase {...props} aria-label="test textarea" />);
+
+    userEvent.type(screen.getByLabelText('test textarea'), 'new test textarea');
+    expect(props.onChange).toHaveBeenCalled();
```
[the remark this answers is not in the document]
Comment (jeffpuzzo) [reply packages/react-core/src/components/TextArea/__tests__/TextArea.test.tsx]: Updated this and the other tests mentioned to better reflect what was previously tested.

**Review by thatblindgeye — APPROVED** [review]

**Review by tlabaj — APPROVED** [review]

**Review by wise-king-sullyman — APPROVED** [review]
🥳

**Pull request merged**